### PR TITLE
Create README files at domain and root level also

### LINKED ISSUE #39: Create root level and domain level README files
Currently, the script generates README files at the sub-domain level (Eg: Hackerrank/Algorithms/BitManipluation/). It'll be good if the script can similarly generate README files at root level (Hackerrank/) and domain level (Eg: Hackerrank/Algorithms/) as well.

### PR DESCRIPTION
**Pull request by rajatgoyal715**
Closes #39 

#### Changelogs
- Create README files at the domain level and root level
- Refactoring current code to make it more readable

### CODE REVIEW

**Comment by rajat19** [comment]
Some checks failed
Please look into it

**Comment by rajatgoyal715** [comment]
> Some checks failed
> Please look into it

Looking into these!

File: hsc/crawler.py [hunk 1]
```diff
@@ -195,47 +168,49 @@ def get_submissions(self, submissions):
...
+					subdomain_name = track['name'].strip().replace(' ', '')
+					domain_slug = track['track_slug']
+					subdomain_slug = track['slug']
+				
+				domain_url = self.domain_url.format(domain_slug)
+				subdomain_url = self.subdomain_url.format(domain_slug, subdomain_slug)
 
-				if self.make_language_folder:
-					folder_name = os.path.join(folder_name, language)
+				# if self.make_language_folder:
```
Comment (rishabhsingh971) [inline hsc/crawler.py]: should specify reason for commenting this

File: hsc/crawler.py [hunk 2]
```diff
@@ -149,27 +80,69 @@ def store_submission(self, file_name, code):
...
-
-	def get_file_path(self, folder_name, file_name_with_extension):
-		return os.path.join(self.base_folder_name, folder_name, file_name_with_extension)
+	def write(self, file_name, text):
+		os.makedirs(os.path.dirname(file_name), exist_ok=True)
+		with open(file_name, 'w') as text_file:
+			text_file.write(text)
+
+
+	def create_readmes(self, domain_name, subdomain_name, domain_url, subdomain_url, subdomain_readme_path, domain_readme_path, root_readme_path):
```
Comment (rishabhsingh971) [inline hsc/crawler.py]: too many arguments.. can it be reduced? 
If not at least consider breaking this line into multiple

File: hsc/crawler.py [hunk 3]
```diff
@@ -149,27 +80,69 @@ def store_submission(self, file_name, code):
...
+		if not os.path.exists(domain_readme_path):
+			text = self.domain_readme_text.format(domain_name, domain_url)
+			self.write(domain_readme_path, text)
+
+		if not os.path.exists(root_readme_path):
+			text = self.root_readme_text.format(self.base_url)
+			self.write(root_readme_path, text)
+
+
+	def update_readmes(self, domain_name, subdomain_name, domain_url, subdomain_url, challenge_name, challenge_slug, language, file_name_with_extension):
```
Comment (rishabhsingh971) [inline hsc/crawler.py]: too many arguments.. can it be reduced? 
If not at least consider breaking this line into multiple

**Review by rishabhsingh971 — CHANGES_REQUESTED** [review]

File: hsc/crawler.py (hunk 1 above)
Comment (rajatgoyal715) [reply hsc/crawler.py]: We weren't using this anyways. Can you please help me out here if we are going to use this in the future?

Comment (rishabhsingh971) [reply hsc/crawler.py]: We haven't exposed this functionality yet. It was to be exposed in #24 as mentioned [here](https://github.com/Nullifiers/Hackerrank-Solution-Crawler/pull/19#issuecomment-518015717). But we forgot to add it. >.<

Comment (rajatgoyal715) [reply hsc/crawler.py]: Fixed this in 14f41faa1928d533b442135b487a00c7d2911a8a

File: hsc/crawler.py (hunk 2 above)
Comment (rajatgoyal715) [reply hsc/crawler.py]: Fixed in 9e7e65c

File: hsc/crawler.py (hunk 3 above)
Comment (rajatgoyal715) [reply hsc/crawler.py]: Fixed in 9e7e65c

**Review by rishabhsingh971 — APPROVED** [review]

**Pull request merged**